Engine/Controllers/RoomController.cs

```diff
@@ -180,5 +167,15 @@ public void InspectRoom()
         {
             currentRoom.Inspect();
         }
+
+        // Will search all gameobject in a room to find match including the room itself.
+        public GameObject GetGameObject(string name)
+        {
+            List<GameObject> searchList = new List<GameObject>();
+            searchList.AddRange(this.currentRoom.getItems());
+            searchList.AddRange(this.currentRoom.getNPCs());
+            return searchList.FirstOrDefault(x => x.Name == name);
```

Comment:
You're creating and adding to a list for no reason. Makes no difference in this case, but the same code with a large number of objects or very high-performance requirements would be suboptimal. You can replace it with:

```
public GameObject GetGameObject(string name) =>
	currentRoom.getItems().Concat(currentRoom.getNPCs()).FirstOrDefault(x => x.Name == name);
```